app/src/main/java/com/gh4a/utils/rx/RxTools.java

```diff
@@ -0,0 +1,102 @@
+package com.gh4a.utils.rx;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableTransformer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import android.app.Activity;
+import android.app.Dialog;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import com.gh4a.BasePagerActivity;
+import com.gh4a.R;
+import com.gh4a.utils.UiUtils;
+import com.philosophicalhacker.lib.RxLoader;
+
+public class RxTools {
+    public static <T> ObservableTransformer<T, T> handle(Activity activity, int id) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+            return observable
+                    .compose(handleError(activity))
+                    .compose(rxLoader.makeObservableTransformer(id));
+        };
+    }
+
+    public static <T> ObservableTransformer<T, T> handle(Activity activity, int id, boolean refresh) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+
+            if(refresh) {
+                return observable
+                        .compose(handleError(activity))
+                        .compose(rxLoader.makeObservableTransformer(id, refresh));
+            } else return observable.compose(handle(activity, id));
+        };
+    }
+
+    // Error handler showing SnackBar
+    public static <T> ObservableTransformer<T, T> onErrorSnackbar(Activity activity, int id, View rootLayout, String errorMessage, int messageRes) {
```

Comment:
As above: pass in BaseActivity and add a Fragment variant. This allows calling activity.getRootLayout() from here without the need to pass it in.

Also the name (and comment) is off: it implies it (only) shows a Snackbar on error, but does much more. It's probably better to split the dialog and snackbar transformers.